### DISCUSSION TITLE
ui.video does not play YouTube link

### DISCUSSION THREAD

**Discussion by wgong** [zauberzeug/nicegui#1673] [Q&A] [answered]
### Question

Try to use ui.video to play YouTube link in below code snippet
```py
from nicegui import ui
url_video = "https://www.youtube.com/watch?v=HljSXSm6v9M"
ui.link(f"YouTube Video {url_video}", url_video)
v = ui.video(url_video)
ui.run()
```
But control is not enabled, is it because url does not end with ".mp4"? any fix or workaround? 
Thanks

**Comment by rodja** [comment, marked as answer]
`ui.video` just wraps a html `<video>`-tag which does not know how to play YouTube links. To embed a YouTube video just use `ui.html` and the iframe provided when clicking on the share link of a video:

```py
ui.html('''
    <iframe width="560" height="315" 
            src="https://www.youtube.com/embed/odyP8VDTfiw?si=69KZmLRgZpxVh7DE" 
            title="YouTube video player" frameborder="0" 
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" 
            allowfullscreen>
    </iframe>''')
```

If you want to control playback speed and the like, you need to create a Player object:

```py
@ui.page('/')
async def index():
    ui.add_head_html('<script src="https://www.youtube.com/iframe_api"></script>')
    ui.add_body_html(r'''
        <div id="player"></div>
        <script>
            var player;
            function onYouTubeIframeAPIReady() {
                player = new YT.Player('player', {
                    height: '390',
                    width: '640',
                    videoId: 'B6kWVSpkmJU'
                });
            }
        </script>
        ''')

    with ui.row():
        ui.button('slow', on_click=lambda: ui.run_javascript('player.setPlaybackRate(0.5);', respond=False))
        ui.button('normal', on_click=lambda: ui.run_javascript('player.setPlaybackRate(1);', respond=False))
        ui.button('fast', on_click=lambda: ui.run_javascript('player.setPlaybackRate(2);', respond=False))
```

This was originally asked [on Discord](https://discord.com/channels/1089836369431498784/1148398798662541463/1148842161530470451).

**Reply by wgong** [reply]
Thank you @rodja , I like the 2nd snippet because I can pass in vid as param, Should we add this as a video example?

**Reply by rodja** [reply]
Good Idea @wgong. I would be happy if you create a pull request.

**Comment by ed2050** [comment]
I noticed the raw `<video>` tag in page source.  Quasar provides a media player component, [see here](https://github.com/quasarframework/quasar-ui-qmediaplayer).  It supports multiple tracks, keyboard shortcuts (see [native controls](https://qmediaplayer.netlify.app/examples/video-native-controls)), mobile players, multiple languages, and more features.

Any thoughts on upgrading the video player in nicegui with this richer component?

**Reply by falkoschindler** [reply]
Interesting, @ed2050! This might be a nice addition in the future. As far as I can tell, it would create a new dependency, since QMediaPlayer is not part of the core Quasar library. So it's a larger undertaking than integrating other Quasar components. But feel free to create a feature request for it to discuss further details.

**Reply by ed2050** [reply]
Done, see #1995.  Dependencies seem minimal, given it's developed by quasar devs and installs as a quasar extension.  But we'll see.  Thanks.
